Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM ubuntu:latest
```

Comment:
Nope, this is only the base image. It works on any OS that can run Docker.

Docker works by using features of OS that allow it to isolate and virtualize resources. It's kinda similar to what VM's does... 

Well, you're running on your OS [MacOS] that runs a VM [i.e:VBox] which has an installed OS [tinyOS] with the needed capabilities (such as isolated process/io/network management). 

On top of this, you'll create a container based on an image (also compatible with that capabilities), in the end the image will provide "just" binaries and folder..

this was a silly, roughly and non-educative introduction to Docker :disappointed: 

For a better understanding you can watch:

* https://sysadmincasts.com/episodes/14-introduction-to-linux-control-groups-cgroups 
* https://sysadmincasts.com/episodes/24-introduction-to-containers-on-linux-using-lxc
* https://sysadmincasts.com/episodes/31-introduction-to-docker